### PR TITLE
[Configurable Entities] Add virtual metadata fields for journals

### PR DESCRIPTION
**Pull request by artlowel**
This PR depends on #315

- On simple item pages for Journal Issues, it adds the following metadata fields to the left column:
	- Journal Title (`journal.title`)
	- Journal ISSN (`journal.identifier.issn`)
- On simple item pages for Journal Articles  it adds the following metadata fields to the left column:
	- Journal Title (`journal.title`)
	- Journal ISSN (`journal.identifier.issn`)
	- Volume title (`journalvolume.identifier.name`)
- On simple item pages for Journals it adds a search component to the bottom of the page that allows you to search or browse through all articles in that journal

Dependent on this REST API PR: https://github.com/DSpace/DSpace/pull/2270
Example: https://dspace7-entities.atmire.com/items/a23eae5a-7857-4ef9-8e52-989436ad2955

### CODE REVIEW

**Comment by AlexanderS** [comment]
(Note: Just to help reviewing this, here is the diff between #315 and this PR: [diff](https://github.com/atmire/dspace-angular/compare/configurable_entities...add-virtual-metadata-fields-for-journals))

**Comment by tdonohue** [comment]
@artlowel : Now that #315 (which this PR is based on) has been merged, could this PR be rebased? It'd make it much easier to review (as it currently seems to contain all the #315 changes as well).

**Comment by paulo-graca** [comment]
I've saw the commits, the code looks ok to me, I understand what changed, but I want to test it.

**Comment by artlowel** [comment]
@tdonohue Done

File: src/app/+item-page/simple/related-entities/related-entities-search/related-entities-search.component.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { Item } from '../../../../core/shared/item.model';
+import { SearchFixedFilterService } from '../../../../+search-page/search-filters/search-filter/search-fixed-filter.service';
+import { isNotEmpty } from '../../../../shared/empty.util';
+import { of } from 'rxjs/internal/observable/of';
+
+@Component({
+  selector: 'ds-related-entities-search',
+  templateUrl: './related-entities-search.component.html'
+})
+export class RelatedEntitiesSearchComponent implements OnInit {
```
Comment (tdonohue) [inline src/app/+item-page/simple/related-entities/related-entities-search/related-entities-search.component.ts]: New Component should have TypeDocs describing it.  (Thanks for the inline comments below though!)

**Comment by paulo-graca** [comment]
I've tried to test this pull request, but I think it's dependent on other pull requests. My installation it's showing errors:

`org.dspace.app.rest.exception.InvalidSearchFilterException: isAuthorOfPublication is not a valid search filter`

The requested URL:
https://hostname/spring-rest/api/discover/facets/entityType?page=0&size=5&sort=score,DESC&page=0&size=10&f.isAuthorOfPublication=8f1cdd60-fdfb-4ec5-bcd8-9c8bb546d83b

I will try to understand if it's related with  DSpace/DSpace#2270 that is already merged.

**Comment by artlowel** [comment]
@paulo-graca Yes you need that PR, and perhaps you're missing a DiscoverySearchFilter config, such as the one [here](https://github.com/atmire/DSpace/blob/atmire-demo/dspace/config/spring/api/discovery.xml#L449)?

**Comment by artlowel** [comment]
Clicked the wrong button by mistake :man_facepalming:

**Comment by paulo-graca** [comment]
Yes @artlowel  the problem was with the missing beans. Thank you! Are those changes available in any Pull Request?

**Comment by artlowel** [comment]
Not yet, but I've notified @benbosman. He'll create one

**Comment by benbosman** [comment]
@paulo-graca and @artlowel I've created the missing PR with the applicable discovery configuration at https://github.com/DSpace/DSpace/pull/2324
This should be a minor config change to merge.

**Comment by benbosman** [comment]
Thanks @paulo-graca 
We have done some research on useful metadata fields for the various objects, and schema.org has a whole list of relevant schemas such as https://schema.org/Person and https://schema.org/Organization
We would like to bring this up as a discussion point in the meetings once the merge in master is completed.

**Pull request merged**